lib/optimizely.rb

```diff
@@ -64,17 +64,15 @@ def initialize(datafile, event_dispatcher = nil, logger = nil, error_handler = n
 
       begin
         @config = ProjectConfig.new(datafile, @logger, @error_handler)
-      rescue
+      rescue InvalidDatafileVersionError => e
```

Comment:
Same comment as on the Python PR, please consolidate into one `rescue` and change the message based on the type of the exception